### PR TITLE
Refactor composing model arch

### PR DESCRIPTION
**Pull request by nv-braf**
Re-architecture to combine BLS & Ensemble composing model configs into a single composing config list.
Lots of refactoring to clean up code/methods, and encapsulate some of the complexity that is involved with BLS and ensemble (especially around config generation).

### CODE REVIEW

File: model_analyzer/config/generate/run_config_generator_factory.py [hunk 1]
```diff
@@ -70,20 +70,16 @@ def create_run_config_generator(
...
-        ensemble_composing_models = RunConfigGeneratorFactory._create_ensemble_composing_models(
+        composing_models = RunConfigGeneratorFactory._create_composing_models(
             new_models, command_config, client, gpus)
 
-        bls_composing_models = RunConfigGeneratorFactory._create_bls_composing_models(
-            command_config, client, gpus)
-
         if (command_config.run_config_search_mode == "quick" or
-                bls_composing_models):
+                composing_models):
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/run_config_generator_factory.py:77]: is this needed? Is there not protection elsewhere for composing models with !quick?

File: model_analyzer/config/run/model_run_config.py [hunk 2]
```diff
@@ -118,10 +110,11 @@ def representation(self) -> str:
 
         return self.perf_config().representation()
 
-    def _check_for_client_vs_model_batch_size(self):
+    def _check_for_client_vs_model_batch_size(self) -> bool:
         """
         Returns false if client batch size is greater than model batch size. Else true
         """
+        # FIXME: should this be checking composing configs?
```
Comment (tgerdesnv) [inline model_analyzer/config/run/model_run_config.py:117]: fixme?

**Review by tgerdesnv — COMMENTED** [review]
Changes like this are where I wish we had L0 tests to protect both ensemble and BLS. Have you manually re-run them?

File: model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py [hunk 3]
```diff
@@ -64,10 +63,8 @@ def __init__(self, search_config: SearchConfig,
         gpus: List of GPUDevices
         models: List of ModelProfileSpec
             List of models to profile
-        ensemble_composing_models: Dict of List of ModelProfileSpec
-            Dict indexed by model name of list of composing models to profile
-        bls_composing_models: List of ModelProfileSpec
-            List of BLS composing models to profile
+        composing_models: List of ModelProfileSpec
+            List of composing models to profile
```
[the remark this answers is not in the document]
Comment (nv-braf) [reply model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py]: Fixed

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**